l3kernel/testfiles/m3coffins001.tlg

```diff
@@ -402,19 +402,23 @@ Defining \2 on line ...
 ....\kern -0.5
 ....\hbox(1.0+0.0)x1.0, shifted 0.5
 .....\pdfcolorstack 0 push {0 g 0 G}
+.....\pdfcolorstack 0 push {1.0 0.0 0.0 rg 1.0 0.0 0.0 RG}
 .....\hbox(1.0+0.0)x1.0
 ......\rule(1.0+*)x1.0
 .....\pdfcolorstack 0 pop
+.....\pdfcolorstack 0 pop
 ...\kern -90.54457
 ...\kern -11.61815
 ...\hbox(3.75+1.25)x9.61815, shifted 5.75
 ....\pdfcolorstack 0 push {0 g 0 G}
+....\pdfcolorstack 0 push {1.0 0.0 0.0 rg 1.0 0.0 0.0 RG}
```

Comment:
why all these double color pushs and pops?

Reply:
The coffins code had a 'conditional' use of colour for coffin pole marking (in the debugging features). Now we have colour support out of the box, they automatically get marked in blue. That happens inside a coffin, which therefore starts with `\color_ensure_current:`, surrounding black, so here `0 g 0 G`) then we apply blue in RGB, so we see that appear. (I was slightly caught out myself until I looked back and remembered how the coffin pole display part was set up.)

Reply:
So couldn't the ensure black go out there or is it needed when the debugging is turned off?

Reply:
I think I wasn't clear :)

At the basic level, we have
```latex
\cs_new_protected:Npn \hcoffin_set:Nn #1#2
  {
    \@@_if_exist:NT #1
      {
        \hbox_set:Nn #1
          {
            \color_ensure_current:
            #2
          }
        \@@_update:N #1
      }
  }
```
which means that coffins work like LaTeX2e user boxes and save the prevailing colour to the start of the underlying TeX box construct. We then have `\coffin_mark_handle:Nnnn`, which takes as arguments a coffin, two poles and _a colour_. Handles are marked by placing their 'name' (the two poles) in a coffin and attaching it in the right place. That text is coloured, so we have in 2e terms
```latex
\hbox{%
   \color@setgroup
   \color{<whatever>}%
   (<pole1>,<pole2>)
   \color@endgroup
}
```
which means we have two colours applied. The only way to avoid that would be to use a `expl3` box not a coffin here, as they don't include `\color_ensure_current:` at present.

Reply:
ok fine